### PR TITLE
Fixed:cssselect extraction failed &&  Parse XPath result in string but not list.

### PR DESCRIPTION
**Pull request by howie6879**
https://github.com/gaojiuli/toapi/commit/30d983eb4aba5fbb2d7ffa7bb13f923bc900cbc9

This commit will cause cssselect extraction failed

The xpath extraction can be solved this way https://github.com/gaojiuli/toapi/issues/1

### CODE REVIEW

**Comment by howie6879** [comment]
image

File: toapi/item.py [hunk 1]
```diff
@@ -26,9 +26,12 @@ def parse(cls, html):
             item = {}
             for name in cls.selectors:
                 try:
-                    item[name] = cls.selectors[name].parse(section)[0]
+                    item[name] = cls.selectors[name].parse(section)
```
Comment (gaojiuli) [inline toapi/item.py]: Why not get string result in this way?

File: toapi/selector.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def parse(self, html):
             # TODO
             value = value[0].get(self.attr).strip() if len(value) == 1 else value
         else:
-            if len(value) == 1:
+            if isinstance(value, list) and len(value) == 1 and isinstance(value[0], etree._Element):
```
Comment (gaojiuli) [inline toapi/selector.py]: Your solution makes thing complex. I just want to get the first item of the XPath result list. Css selector still not works well.

Comment (howie6879) [reply toapi/selector.py]: I have already sent an email to give you an explanation.

**Pull request merged**